### PR TITLE
#167783727 View User Profile

### PR DESCRIPTION
**Pull request by JuwonAbiola**
#### What does this PR do?
- It enables a user to view other users profile

#### Description of Task proposed in this pull request?
- Create a controller method that views users profile
#### How should this be manually tested (Quality Assurance)?
- on your postman make a GET request to /profiles/:id

#### What are the relevant pivotal tracker stories?
- [167783727](https://www.pivotaltracker.com/story/show/167783727)

#### Any background context you want to add (Operations Impact)?
- N/A

#### What I have learned working on this feature:
- I learnt how to stub using sandbox

#### Screenshots: 
image

### CODE REVIEW

File: server/docs/authors-haven-api.yml [hunk 1]
```diff
@@ -364,6 +364,32 @@ paths:
           description: Success. Email verification was successful
         500:
           description: Internal server errorcomponents     
+  /api/v1/users/profile:
```
Comment (chialuka) [inline server/docs/authors-haven-api.yml]: It appears this route is incorrect. The `routes/profiles.js` page requires a param to work. This returns a 404

File: server/docs/authors-haven-api.yml [hunk 2]
```diff
@@ -364,6 +364,32 @@ paths:
...
+      security:
+        - Authorization: []
+      responses:
+        200:
+          description: user fetched successfully
+        404:
+          description: user does not exist
+        500:
+          description: Internal server error
+  /api/v1/users/profile/2:
```
Comment (chialuka) [inline server/docs/authors-haven-api.yml]: Is the profiles route mounted on `/users/profiles`? From your routes, it is accessed via `/profiles`. The route in this documentation does not exist

Comment (JuwonAbiola) [reply server/docs/authors-haven-api.yml]: Thanks for the feedback but this PR is still in progress

**Review by ayodejiAA — DISMISSED** [review]

File: server/controllers/Profiles.js [hunk 3]
```diff
@@ -19,7 +19,10 @@ class Profiles {
   static async update(req, res) {
     try {
       const {
-        body, user, file, params
+        body,
```
Comment (dinobi) [inline server/controllers/Profiles.js]: why breaking into new lines?

File: server/controllers/Profiles.js [hunk 4]
```diff
@@ -19,7 +19,10 @@ class Profiles {
   static async update(req, res) {
```
Comment (dinobi) [inline server/controllers/Profiles.js]: Please undo all the changes in this method. Sad that I have to keep talking about this

File: server/controllers/Profiles.js [hunk 5]
```diff
@@ -68,6 +68,34 @@ class Profiles {
...
+        params: { id }
+      } = req;
+      const userProfile = await User.findById(id);
+      if (!userProfile) {
+        return serverResponse(res, 404, {
+          error: 'user does not exist'
+        });
+      }
+      delete userProfile.dataValues.password;
+      return serverResponse(res, 200, {
```
Comment (dinobi) [inline server/controllers/Profiles.js]: can we create userResponse helper thats basically works like serverResponse, but have an added responsibility of deleting the password from the user project before sending to client?

File: server/database/migrations/20190813101408-create-role.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+export default {
```
Comment (dinobi) [inline server/database/migrations/20190813101408-create-role.js]: Why do we have a this in this PR?

File: server/controllers/Profiles.js (hunk 4 above)
Comment (JuwonAbiola) [reply server/controllers/Profiles.js]: The changes occurred due to merge conflicts

**Review by henryade — DISMISSED** [review]

**Review by chialuka — DISMISSED** [review]

File: server/controllers/Profiles.js [hunk 7]
```diff
@@ -68,6 +73,29 @@ class Profiles {
...
+  static async view(req, res) {
+    try {
+      const {
+        params: { id }
+      } = req;
+      const userProfile = await User.findById(id);
+      if (!userProfile) {
+        return serverResponse(res, 404, { error: 'user does not exist' });
+      }
+      return userResponse(res, 200, userProfile);
```
Comment (dinobi) [inline server/controllers/Profiles.js:91]: Please use this in other places where we delete password as well.

**Review by dinobi — DISMISSED** [review]

**Review by chialuka — APPROVED** [review]

**Review by abdulfatai360 — APPROVED** [review]

**Review by Adekoreday — APPROVED** [review]

**Review by henryade — APPROVED** [review]

**Review by dinobi — APPROVED** [review]

**Pull request merged**